### PR TITLE
hack/get-terraform: Use 'go env ...' to support other platforms

### PR DESCRIPTION
**Pull request by wking**
With this change, we use Go to give the correct OS and architecture constants for the Terraform URL.  Go, because Terraform is a Go project, so Hashicorp is building the binaries using Go's cross-compilation with the Go naming.

If you don't have Go, I'm just assuming you're on amd64 Linux, because
that's what our CI infrastructure is running.

### CODE REVIEW

**Comment by ironcladlou** [comment]
I'm thinking `uname` would satisfy the POSIX gods

**Comment by rphillips** [comment]
+1 to using uname... `set -e` and we can remove all the && as well.

File: hack/get-terraform.sh [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+	OS="$(go env GOOS)" &&
+	ARCH="$(go env GOARCH)"
+fi &&
 TERRAFORM_VERSION="0.11.8" &&
-TERRAFORM_URL="https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip" &&
+TERRAFORM_URL="https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_${OS}_${ARCH}.zip" &&
+echo "pulling ${TERRAFORM_URL}" >&2 &&
 cd "$(dirname "$0")/.." &&
 mkdir -p bin &&
 curl -L "${TERRAFORM_URL}" | funzip > bin/terraform &&
```
Comment (mike-nguyen) [inline hack/get-terraform.sh]: gunzip?

Comment (wking) [reply hack/get-terraform.sh]: > s/funzip/gunzip ?

```console
$ wget https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip
$ file terraform_0.11.8_linux_amd64.zip 
terraform_0.11.8_linux_amd64.zip: Zip archive data, at least v2.0 to extract
```

`funzip` is about extracting files from ZIP archives (what we want here).  `gunzip` is about removing gzip compression.  It looks like they both function equivalently for this single-file ZIP, but `funzip` is a more accurate semantic match (I can't find docs for `gunzip` being able to handle ZIP archives, what happens if Hashicorp adds another file to the zip?).

Did you have specific reasons for changing this?

Comment (mike-nguyen) [reply hack/get-terraform.sh]: I was running through the installer and didn't have funzip installed and couldn't find it in the fedora repos.  Your reasoning for using funzip makes sense but maybe we need a note saying it is a dependency? (unless I missed it somewhere).  It was easy enough to workaround it.

**Comment by wking** [comment]
> I'm thinking `uname` would satisfy the POSIX gods

But then we'd need local logic to translate to the Go identifiers.  If we're calling an external executable, why not call `go` and get the names without needing to translate?

Also, I expect Windows devs are more likely to have a working `go` than they are to have a working `uname` (although they may not have a working `sh` to run the script, in which case, this point doesn't matter).

> `set -e` and we can remove all the `&&` as well.

I slightly prefer `&&` chaining (maybe I spent too much time reading [the Git test suite][1] ;).  But I'm fine changing it if other folks prefer `set -e`.  Thoughts?

[1]: https://github.com/git/git/blob/master/t/t0000-basic.sh#L38-L46

File: hack/get-terraform.sh (hunk 1 above)
Comment (wking) [reply hack/get-terraform.sh]: > I was running through the installer and didn't have funzip installed...

I've pushed a belt-and-suspenders 78bf9f7dc ;).

> ... and couldn't find it in the fedora repos...

I expect it's in `unzip` based on [this old `Dockerfile`][1].

[1]: https://github.com/openshift/installer/commit/64e02b3be1cae3d311feb48f41152a7b1f8bf305#diff-e93aaab1fdfc92bac48708be1c4df739L35

**Pull request merged**